CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,345 @@
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+
+#include "CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.h"
+
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+
+const int EcalEBPhase2TPParamProducer::linTopRange_ = 16383;
+
+EcalEBPhase2TPParamProducer::EcalEBPhase2TPParamProducer(edm::ParameterSet const& pSet)
+    : theBarrelGeometryToken_(esConsumes(edm::ESInputTag("", "EcalBarrel"))),
+      inFile_(pSet.getUntrackedParameter<std::string>("inputFile")),
+      outFile_(pSet.getUntrackedParameter<std::string>("outputFile")),
+      nSamplesToUse_(pSet.getParameter<unsigned int>("nSamplesToUse")),
+      useBXPlusOne_(pSet.getParameter<bool>("useBXPlusOne")),
+      phaseShift_(pSet.getParameter<double>("phaseShift")),
+      nWeightGroups_(pSet.getParameter<unsigned int>("nWeightGroups")),
+      theEcalTPGPedestals_Token_(esConsumes(edm::ESInputTag("EcalLiteDTUPedestals", ""))),
+      et_sat_(pSet.getParameter<double>("Et_sat")),
+      xtal_LSB_(pSet.getParameter<double>("xtal_LSB")),
+      binOfMaximum_(pSet.getParameter<unsigned int>("binOfMaximum"))
+
+{
+  out_file_ = gzopen(outFile_.c_str(), "wb");
+
+  TFile* inFile = new TFile(inFile_.c_str(), "READ");
+
+  inFile->GetObject("average-pulse", thePulse_);
```

Comment:
Hi Nancy,

Thanks for implementing "LMT/LMF" for phase II TPs. I just have a couple questions. First, here, why are you accessing the pulse shape from some file? The DIGI code we already have for phase 2 digis accesses the pulse shape from the database -- you can see the pulse shape here:
https://cern.ch/pgsvb
Wouldn't it improve the consistency and maintainability of the code to do the same here?

(comments continued elsewhere)

Reply:
@jhakala n More detailed comments here.  The file we are accessing the pulse shape is built on the base of the pulse implemented in CMSSW. Just a matter of implementation, no changes about the neat info. We will move, at a later stage, to a better implementation which will not need the intermidiate files